habitat/core/dataset.py

```diff
@@ -355,9 +367,65 @@ def _shuffle_iterator(self) -> None:
                 list(g)
                 for k, g in groupby(self._iterator, key=lambda x: x.scene_id)
             ]
-            random.shuffle(grouped_episodes)
+            grouped_episodes += []
+
+            for i in range(len(grouped_episodes)):
+                random.shuffle(grouped_episodes[i])
+
+            if len(grouped_episodes) > 1:
+                # Ensure we swap episodes by moving the current group to the end
+                random.shuffle(grouped_episodes[1:])
+                grouped_episodes = grouped_episodes[1:] + grouped_episodes[0:1]
+
             self._iterator = iter(sum(grouped_episodes, []))
         else:
             episodes = list(self._iterator)
             random.shuffle(episodes)
             self._iterator = iter(episodes)
+
+    def step_taken(self):
+        self._step_count += 1
+
+    @staticmethod
+    def _randomize_value(value, interval):
+        return random.randint(
+            int(value * (1 - interval)), int(value * (1 + interval))
+        )
+
+    def _set_shuffle_intervals(self):
+        if self.max_scene_repetition_episodes > 0:
+            self._max_rep_episode = self._randomize_value(
```

Comment:
I'm not sure if we need to randomize the episode rep count. I understand that it is helpful for step rep count but it's kind of counter-intuitive for episode rep count, with a default value of `0.2`.

Reply:
Makes sense to rename it and not do it the episode count method.